src/test/java/eu/europa/ted/efx/sdk2/EfxExpressionTranslatorV2Test.java

```diff
@@ -87,10 +87,34 @@ void testLikePatternCondition_WithNot() {
         "BT-00-Text", "'123' not like '[0-9]*'");
   }
 
+  @Test
+  void testLikePatternCondition_WithTextField() {
+    testExpressionTranslation("fn:matches(normalize-space(PathNode/TextField/normalize-space(text())), '[0-9]*')",
+        "{ND-Root} ${BT-00-Text like '[0-9]*'}");
+  }
+
+  @Test
+  void testLikePatternCondition_WithTextMultilingualField() {
+    testExpressionTranslation("every $lang in PathNode/TextMultilingualField/@languageID satisfies fn:matches(normalize-space(PathNode/TextMultilingualField[./@languageID = $lang]/normalize-space(text())), '[0-9]*')",
+        "{ND-Root} ${every text:$lang in BT-00-Text-Multilingual/@languageID satisfies BT-00-Text-Multilingual[BT-00-Text-Multilingual/@languageID == $lang]  like '[0-9]*'}");
+  }
+
+  @Test 
+  void testVisualisationLanguageFunction() {
+    testExpressionTranslation("PathNode/TextMultilingualField[./@languageID = ((for $__LANG__ in ('eng') return if (.[@languageID=$__LANG__]/normalize-space(text())) then $__LANG__ else ()), ./normalize-space(text()))[1]]/normalize-space(text())", 
```

Comment:
I would expect this to use the "ted:preferred-languages" function, as commented in XpathScriptGenerator.java.